File: WorldFeed/Server/src/WorldFeed.Common/Public/Events/DisconnectedEventArgs.cs
namespace WorldFeed.Common.Public.Events
{
    using System;

    using WorldFeed.Common.Public.Streaming.Events;

    /// <summary>
    /// The stream was disconnected
    /// </summary>
    public class DisconnectedEventArgs : EventArgs
    {
        public DisconnectedEventArgs(IDisconnectMessage disconnectMessage)
        {
            DisconnectMessage = disconnectMessage;
        }

        public IDisconnectMessage DisconnectMessage { get; }
    }
}

File: Sharebook/Client/src/app/core/Public/Events/DisconnectedEventArgs.ts
import {IDisconnectMessage} from "../Streaming/Events/IDisconnectMessage";

// The stream was disconnected
export class DisconnectedEventArgs /*extends EventArgs*/ {
  constructor(disconnectMessage: IDisconnectMessage) {
    // super();

    this.DisconnectMessage = disconnectMessage;
  }

  public DisconnectMessage: IDisconnectMessage;
}
